metaflow/plugins/conda/conda_step_decorator.py

```diff
@@ -170,10 +176,21 @@ def _prepare_step_environment(self, step_name, ds_root):
                               env_id,
                               cached_deps[env_id]['urls'],
                               architecture=self.architecture,
-                              explicit=True)
-            CondaStepDecorator.environments = CondaStepDecorator.conda.environments(self.flow.name)
+                              explicit=True,
+                              disable_safety_checks=self.disable_safety_checks)
+            CondaStepDecorator.environments =\
+                CondaStepDecorator.conda.environments(self.flow.name)
         return env_id
 
+    def _disable_safety_checks(self, decos):
+        # Disable conda safety checks when creating linux-64 environments on
+        # a macOS. This is needed because of gotchas around inconsistently 
+        # case-(in)sensitive filesystems for macOS and linux.
+        for deco in decos:
+            if deco.name == 'batch' and platform.system() == 'Darwin':
```

Comment:
Do we want to disable safety checks for all steps including the non-batch step? Users can create a metaflow conda environment running locally and then run --with batch. Would it be a problem if the env created in non-batch mode is then directly used for batch run?

Reply:
This only happens when we resolve linux-64 packages on mac. Currently we create a different environment for batch and non-batch mode when launched from mac. I hope this clarifies.